helper/resource.js

```diff
@@ -0,0 +1,128 @@
+        }catch(err){
+            res.status(400).send({
+                result: false,
+                message: err.message
+            });
+        }
+    })
+}
+
+//this function will call druid and return all the organisations to particular program or solution
+exports.getOrganisations = async function(req,res){
+    return new Promise(async function (resolve, reject) {
+        try {
+            let bodyParam = req.query.resourceType === ResourceType.SOLUTION ? gen.utils.getDruidQuery("solution_organisations_level_query") : gen.utils.getDruidQuery("program_organisations_level_query");
+            //Gets data source name based on resource type and solution type
+            bodyParam.dataSource = await utils.getDataSourceName(req.query, req.body)
+            //Gets Resource filter based on Resource type
+            const resourceFilter = await utils.getResourceFilter(req.query)
+            bodyParam.filter.fields.push(resourceFilter)
+            let options = gen.utils.getDruidConnection();
+            options.method = "POST";
+            options.body = bodyParam;
+            let data = await rp(options);
+            if(data){
+                const result = data.map(district => ({ id: district.event.organisation_id, name: district.event.organisation_name }));
+                resolve(result)
+            }
+        }catch(err){
+            res.status(400).send({
+                result: false,
+                message: err.message
+            });
+        }
+    })
+
+}
+
+//this function will call druid and return all the block to particular program or solution
+exports.getBlocks = async function(req,res){
+    return new Promise(async function (resolve, reject) {
+        try {
+            let bodyParam = req.query.resourceType === ResourceType.SOLUTION ? gen.utils.getDruidQuery("solution_block_level_query") : gen.utils.getDruidQuery("program_block_level_query");
+            //Gets data source name based on resource type and solution type
+            bodyParam.dataSource = await utils.getDataSourceName(req.query, req.body)
+            //Gets Resource filter based on Resource type
+            const resourceFilter = await utils.getResourceFilter(req.query)
+            bodyParam.filter.fields.push(resourceFilter)
+
+            const districtFilter = {
+                type: "selector",
+                dimension: req.query.resourceType == ResourceType.SOLUTION ?  "district_externalId" : "district_id",
+                value: req.body.query.districtLocationId
+            }
+            bodyParam.filter.fields.push(districtFilter)
+            let options = gen.utils.getDruidConnection();
+            options.method = "POST";
+            options.body = bodyParam;
+            let data = await rp(options);
+            if(data){
+                const result = data.map(block => ({ id: req.query.resourceType == ResourceType.SOLUTION ? block.event.block_externalId : block.event.block_id, name: block.event.block_name }));
```

Comment:
@ankitpws we can avoid checking in loop the same if condition. Also why we need for solution external id and for program the block id? What is the reason